File: frontend/src/types/voice-design.ts
export interface VoiceDesign {
  id: number
  user_id: number
  name: string
  backend_type: 'local' | 'aliyun'
  instruct: string
  aliyun_voice_id?: string
  meta_data?: Record<string, any>
  preview_text?: string
  created_at: string
  last_used: string
  use_count: number
}

export interface VoiceDesignCreate {
  name: string
  instruct: string
  backend_type: 'local' | 'aliyun'
  aliyun_voice_id?: string
  meta_data?: Record<string, any>
  preview_text?: string
}

export interface VoiceDesignListResponse {
  designs: VoiceDesign[]
  total: number
}

File: backend/schemas/voice_design.py
from typing import Optional, Dict, Any, List
from datetime import datetime
from pydantic import BaseModel, Field

class VoiceDesignCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    instruct: str = Field(..., min_length=1)
    backend_type: str = Field(..., pattern="^(local|aliyun)$")
    aliyun_voice_id: Optional[str] = None
    meta_data: Optional[Dict[str, Any]] = None
    preview_text: Optional[str] = None

class VoiceDesignResponse(BaseModel):
    id: int
    user_id: int
    name: str
    backend_type: str
    instruct: str
    aliyun_voice_id: Optional[str]
    meta_data: Optional[Dict[str, Any]]
    preview_text: Optional[str]
    created_at: datetime
    last_used: datetime
    use_count: int

    class Config:
        from_attributes = True

class VoiceDesignListResponse(BaseModel):
    designs: List[VoiceDesignResponse]
    total: int
